### PR TITLE
Characterize - add tristate controls

### PR DESCRIPTION
**Pull request by sven-oly**
This replaces the checkboxes with a tristate control for picking the set of tests to show. These are the options:
"?": does not select the characterization
checkmark: selects the characterization, intersecting with other selected options
"X": selects the characterization whose tests should be omitted from the set to display.

For example, for the following selections
- checkmark for characterization A
- checkmark for characterization C
- X for characterization D
the result is the the tests that are both in A and C, but not in D.

This PR also fixes the accidental omission of the timestamp for detailed reports.

### CODE REVIEW

File: verifier/detail_template.html [hunk 1]
```diff
@@ -224,11 +224,41 @@
...
+      // Start the set of selected items.
+      test_data.selected_set = newSet;
+      } else {
+      // Update by intersection with this set.
+      test_data.selected_set =
+      new Set([...test_data.selected_set].filter((x) => newSet.has(x)));
+      }
+      }
+            else if (check_box.checked) {
+               // A checked box.
```
Comment (echeran) [inline verifier/detail_template.html:261]: fix the indentation of this region

Comment (sven-oly) [reply verifier/detail_template.html]: Fixed this. Also removed old code and revised a lot to be clearer. PTAL.

**Comment by sven-oly** [comment]
PTAL. I think I've fixed the sets that are shown with selection of include / exclude.

**Comment by sven-oly** [comment]
I cleaned up some parts of testreport.py to fix the characterizations and make somewhat better python code.

Ready for your review.

File: verifier/detail_template.html [hunk 2]
```diff
@@ -150,145 +155,153 @@
...
+                // Update by removing all these labels, i.e., intersect with the inverse
+                for (let label of label_set) {
+                    excluded_set.add(label);
+                }
+            }
+        }
+        
+        // Remove the excluded values from selected set.
+        excluded_set.forEach((label) => {
+            test_data.selected_set.delete(label);
```
Comment (echeran) [inline verifier/detail_template.html:272]: Comparing this line on 272 to line 260, it seems as if finding the intersection of the checked options is done one way (using `.filter((x) => label_set.has(x))`) that isn't followed when doing the intersection of the negation of the crossed/excluded options (using `.forEach((label) => {test_data.selected_set.delete(label);})`)

So long as it all works correctly, it's fine. And if you're using sets, and your code is always recomputing set intersections whenever the UI is updated after an updated selection of checks and crosses of key-value pairs, then I suppose it should all work.

**Review by echeran — APPROVED** [review]
LGTM

**Pull request merged**